patches/server/0172-PreCreatureSpawnEvent.patch

```diff
@@ -165,12 +173,26 @@ index b2bb9bbd3af414c50ec3f8e3e171a679e95ef75e..5338e0e1a67925da0c386735a545bb31
 +            );
 +            if (!event.callEvent()) {
 +                if (event.shouldAbortSpawn()) {
-+                    return null;
++                    return PreSpawnStatus.ABORT; // Paper
 +                }
-+                return false;
++                return PreSpawnStatus.CANCELLED; // Paper
 +            }
 +        }
 +        // Paper end
          if (entitytypes.getCategory() == MobCategory.MISC) {
-             return false;
+-            return false;
++            return PreSpawnStatus.FAIL; // Paper
          } else if (!entitytypes.canSpawnFarFromPlayer() && squaredDistance > (double) (entitytypes.getCategory().getDespawnDistance() * entitytypes.getCategory().getDespawnDistance())) {
+-            return false;
++            return PreSpawnStatus.FAIL; // Paper
+         } else if (entitytypes.canSummon() && NaturalSpawner.canSpawnMobAt(world, structureAccessor, chunkGenerator, group, spawnEntry, pos)) {
+             SpawnPlacements.Type entitypositiontypes_surface = SpawnPlacements.getPlacementType(entitytypes);
+ 
+-            return !NaturalSpawner.isSpawnPositionOk(entitypositiontypes_surface, world, pos, entitytypes) ? false : (!SpawnPlacements.checkSpawnRules(entitytypes, world, MobSpawnType.NATURAL, pos, world.random) ? false : world.noCollision(entitytypes.getAABB((double) pos.getX() + 0.5D, (double) pos.getY(), (double) pos.getZ() + 0.5D)));
++            return !NaturalSpawner.isSpawnPositionOk(entitypositiontypes_surface, world, pos, entitytypes) ? PreSpawnStatus.FAIL : (!SpawnPlacements.checkSpawnRules(entitytypes, world, MobSpawnType.NATURAL, pos, world.random) ? PreSpawnStatus.FAIL : world.noCollision(entitytypes.getAABB((double) pos.getX() + 0.5D, (double) pos.getY(), (double) pos.getZ() + 0.5D)) ? PreSpawnStatus.SUCCESS : PreSpawnStatus.FAIL); // Paper
```

Comment:
Should just put the boolean into an extra local var first, then return the spawn status, to keep this easier to read (especially on updates)

Reply:
Good idea, done.